intezer_sdk/results.py

```diff
@@ -0,0 +1,104 @@
+
+    def __iter__(self):
+        """Iterate for page."""
+        if not self._current_page:
+            self._fetch_page()
+        yield from self._current_page
+        self._fetch_page()
+        yield from iter(self)
+
+    def current_page(self) -> List:
+        """Get current page, if not exits, ask a new one from server."""
+        return self._current_page or self._fetch_page()
+
+    def all(self) -> List:
+        """Return all exits analysis's from server."""
+        self._fetch_all_pages()
+        self._current_page_number = 0
+        self._current_page = self.pages[0]
+
+        return self._unite_all_pages_to_one(self.pages)
+
+    def previous_page(self) -> List:
+        """Move to the previous page."""
+        if self._current_page_number - 1 >= 0:
+            self._current_page_number -= 1
+            self._current_page = self.pages[self._current_page_number]
+        return self._current_page
+
+    def next_page(self) -> List:
+        """Move to the next page"""
+        if self._current_page_number == len(self.pages) - 1:
+            return self._fetch_page()
+        self._current_page_number += 1
+        self._current_page = self.pages[self._current_page_number]
+        return self._current_page
+
+    def _fetch_page(self) -> List:
+        """Request for new page from server."""
+        self.data['offset'] = self._current_offset
+        self.total_count, new_page = self._fetch_analyses_history(
+            self._url, self.data)
+
+        if not new_page:
+            raise StopIteration()
+        self._current_page = new_page
+        return self._update_current_page()
+
+    def _update_current_page(self) -> List:
+        """Update all metadata about new current page."""
+        self._current_offset += len(self._current_page)
+        self.pages.append(self._current_page)
+        self._current_page_number = len(self.pages) - 1
+        return self._current_page
+
+    def _fetch_all_pages(self):
+        """Request for all missing pages didn't request yet."""
+        while not len(self.pages) or len(self.pages) * self._page_size < self.total_count:
+            try:
+                self._fetch_page()
+            except StopIteration:
+                break
+
+    def _fetch_analyses_history(self, url_path: str, data: Dict
+                                ) -> Tuple[int, List]:
+        """
+        Request from server filtered analyses history.
+        :param url_path: Url to request new data from.
+        :param data: filtered data.
+        :return: Count of all results exits in filtered request and amount analyses as requested.
+        """
+        response = self.api.request_with_refresh_expired_access_token(
+            path=url_path, method='POST', data=data)
+        raise_for_status(response)
+        json_response = response.json()
+        return json_response["total_count"], json_response["analyses"]
+
+    @staticmethod
+    def _unite_all_pages_to_one(all_pages: List):
```

Comment:
I didn't understand what we were doing here, let's talk about it

Reply:
this is a helper function for all that gives you a flat list of all the analyses's. Because the user doesn't care about the pagination so if he\she asks for all the analyses's they should get everything as a flat list.